presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -362,7 +377,44 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
                     conf,
                     typeManager);
         }
-        return new HiveWriter(hiveFileWriter, partitionName, isNew, fileNameWithExtension, write.toString(), target.toString());
+
+        String writerImplementation = hiveFileWriter.getClass().getName();
+
+        // TODO: the lambda takes an ugly parameter
+        Consumer<HiveWriter> onCommit = (HiveWriter hiveWriter) ->
```

Comment:
looks like so; btw the the lambda is kinda ugly given it needs to take a param so that the row number can be accessed. Any better suggestions?